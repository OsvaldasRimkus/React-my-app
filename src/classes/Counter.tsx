import * as React from 'react';

interface CounterState {
    count: number
}

class Counter extends React.Component<{}, CounterState> {

    constructor(props:any){
        super(props);
        this.state = {
            count: 0,
        }
    }

    modify(i:number) {
        this.setState({count: (this.state.count + i)})
    }

    // increment() {
    //     this.setState({count: (this.state.count + 1)})
    // }

    // decrement() {
    //     this.setState({count: (this.state.count - 1)})
    // }

    render() {
        return(
            <div>
                <span>Current count: {this.state.count}</span>
                <div>
                    <button className="increment" onClick={() => this.modify(1)}>Increment count</button>
                    <button className="decrement" onClick={() => this.modify(-1)}>Decrement count</button>
                </div>
            </div>
        ) 
    }
}

export default Counter;