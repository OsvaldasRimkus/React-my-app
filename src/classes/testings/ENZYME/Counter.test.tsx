import * as React from 'react';
import Counter from '../../Counter';
import {shallow} from 'enzyme';
// import {mount} from 'enzyme';
// import {render} from 'enzyme';
import * as renderer from 'react-test-renderer';


// this is this not take from setup.tsx ?
const Enzyme = require('enzyme');
const Adapter = require('enzyme-adapter-react-16');
Enzyme.configure({ adapter: new Adapter() });

describe('Counter component', () => {
    test('starts with a count of 0', () => {
        const wrapper = shallow (<Counter/>)
        const text = wrapper.find('span').text()
        expect(text).toEqual('Current count: 0')
        // const countState = wrapper.state().count
        // expect(countState).toEqual(0)
    })

    test('can increment the count when the button is clicked', () => {
        const wrapper = shallow(<Counter />)
        const incrementButton = wrapper.find('button.increment')
        incrementButton.simulate('click');
        const text = wrapper.find('span').text()
        expect(text).toEqual('Current count: 1')
    })

    test('can decrement the count when the button is clicked', () => {
        const wrapper = shallow(<Counter />)
        const decrementButton = wrapper.find('button.decrement')
        decrementButton.simulate('click');
        const text = wrapper.find('span').text()
        expect(text).toEqual('Current count: -1')
        console.log("Great success!!!");
    })

    // bad
    // test('should shallow correctly', () => {
    //     expect(shallow(<Counter />)).toMatchSnapshot()
    // })

    // test('should mount correctly', () => {
    //     expect(mount(<Counter />)).toMatchSnapshot()
    // })

    // test('should render correctly', () => {
    //     expect(render(<Counter />)).toMatchSnapshot()
    // })

    // good 
    test('matches the best snapshot', () => {
        const tree = renderer.create(<Counter />).toJSON()
        expect(tree).toMatchSnapshot();
        // expect(tree).toMatchInlineSnapshot();
    })

})

// testing with ENZYME
// ENZYME tests React component functionality
// https://javascriptplayground.com/introduction-to-react-tests-enzyme/

// run with: npm run test