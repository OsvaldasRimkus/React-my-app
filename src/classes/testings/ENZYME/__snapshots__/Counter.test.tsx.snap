// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component matches the best snapshot 1`] = `
<div>
  <span>
    Current count: 
    0
  </span>
  <div>
    <button
      className="increment"
      onClick={[Function]}
    >
      Increment count
    </button>
    <button
      className="decrement"
      onClick={[Function]}
    >
      Decrement count
    </button>
  </div>
</div>
`;
